### ISSUE TITLE
Cant use AVRouteDetector

### ISSUE THREAD

**Issue by gmoney494** [dotnet/macios#10139]
### Steps to Reproduce

1. try to instantiate AVRouteDetector, like `var detector = new AVRouteDetector();`

### Expected Behavior

object to initialize.

### Actual Behavior

get a build error. `CS1729: 'AVRouteDetector' does not contain a constructor that takes 0 arguments`

### Basic Information

- Platform Target Frameworks: 
  - iOS:  Xamarin.iOS 14.6 - saw it in earlier 14.x versions.
- NuGet Packages:
- Affected Devices:

### Environment

```
=== Visual Studio Community 2019 for Mac ===

Version 8.8.1 (build 37)
Installation UUID: 151b314b-22df-4fda-9726-2a61ae79b278
	GTK+ 2.24.23 (Raleigh theme)
	Xamarin.Mac 6.18.0.23 (d16-6 / 088c73638)

	Package version: 612000093

=== Mono Framework MDK ===

Runtime:
	Mono 6.12.0.93 (2020-02/620cf538206) (64-bit)
	Package version: 612000093

=== Roslyn (Language Service) ===

3.8.0-5.20515.7+713efb0e68d34ab75faaf709e071e5c2d8d4ed0e

=== NuGet ===

Version: 5.8.0.6860

=== .NET Core SDK ===

SDK: /usr/local/share/dotnet/sdk/5.0.100/Sdks
SDK Versions:
	5.0.100
	3.1.403
	3.1.402
	3.1.401
	3.1.302
	3.1.301
	3.1.300
	3.1.202
	3.1.200
	3.1.102
	3.1.101
	3.1.100
	3.0.101
	3.0.100
	2.2.203
	2.2.107
	2.1.505
MSBuild SDKs: /Applications/Visual Studio.app/Contents/Resources/lib/monodevelop/bin/MSBuild/Current/bin/Sdks

=== .NET Core Runtime ===

Runtime: /usr/local/share/dotnet/dotnet
Runtime Versions:
	5.0.0
	3.1.9
	3.1.8
	3.1.7
	3.1.6
	3.1.5
	3.1.4
	3.1.2
	3.1.1
	3.1.0
	3.0.1
	3.0.0
	2.2.5
	2.2.4
	2.1.23
	2.1.22
	2.1.21
	2.1.20
	2.1.19
	2.1.18
	2.1.17
	2.1.16
	2.1.15
	2.1.14
	2.1.13
	2.1.9

=== .NET Core 3.1 SDK ===

SDK: 3.1.403

=== Xamarin.Profiler ===

Version: 1.6.12.26
Location: /Applications/Xamarin Profiler.app/Contents/MacOS/Xamarin Profiler

=== Updater ===

Version: 11

=== Apple Developer Tools ===

Xcode 12.2 (17535)
Build 12B45b

=== Xamarin.Mac ===

Xamarin.Mac not installed. Can't find /Library/Frameworks/Xamarin.Mac.framework/Versions/Current/Version.

=== Xamarin.iOS ===

Version: 14.6.0.15 (Visual Studio Community)
Hash: 87a1b18d8
Branch: d16-8
Build date: 2020-11-16 21:39:41-0500

=== Xamarin Designer ===

Version: 16.8.0.507
Hash: e87b24884
Branch: remotes/origin/d16-8
Build date: 2020-10-29 00:31:38 UTC

=== Xamarin.Android ===

Version: 11.1.0.17 (Visual Studio Community)
Commit: xamarin-android/d16-8/c0e2b8e
Android SDK: /Users/a206525411/Library/Developer/Xamarin/android-sdk-macosx
	Supported Android versions:
		7.1 (API level 25)
		8.0 (API level 26)
		8.1 (API level 27)

SDK Tools Version: 26.1.1
SDK Platform Tools Version: 30.0.4
SDK Build Tools Version: 30.0.2

Build Information: 
Mono: be2226b
Java.Interop: xamarin/java.interop/d16-8@79d9533
ProGuard: Guardsquare/proguard/proguard6.2.2@ebe9000
SQLite: xamarin/sqlite/3.32.1@1a3276b
Xamarin.Android Tools: xamarin/xamarin-android-tools/d16-8@2fb1cbc

=== Microsoft OpenJDK for Mobile ===

Java SDK: /Users/a206525411/Library/Developer/Xamarin/jdk/microsoft_dist_openjdk_1.8.0.25
1.8.0-25
Android Designer EPL code available here:
https://github.com/xamarin/AndroidDesigner.EPL

=== Android SDK Manager ===

Version: 16.8.0.32
Hash: 01a7774
Branch: remotes/origin/d16-8
Build date: 2020-10-13 23:32:30 UTC

=== Android Device Manager ===

Version: 16.8.0.45
Hash: fc0af5f
Branch: remotes/origin/d16-8
Build date: 2020-10-13 23:32:54 UTC

=== Build Information ===

Release ID: 808010037
Git revision: acd8ddc086768c701bd641d6ec27316f45813175
Build date: 2020-11-17 12:49:41-05
Build branch: release-8.8
Xamarin extensions: acd8ddc086768c701bd641d6ec27316f45813175

=== Operating System ===

Mac OS X 10.15.7
Darwin 19.6.0 Darwin Kernel Version 19.6.0
    Thu Oct 29 22:56:45 PDT 2020
    root:xnu-6153.141.2.2~1/RELEASE_X86_64 x86_64

=== Enabled user installed extensions ===

DeepClean 1.2.5
Xamarin.Forms HotReload extension 1.4.0

```

**Comment by gmoney494** [comment]
Originally: https://github.com/xamarin/Xamarin.Forms/issues/12949

I started a Xamarin Forum thread post here as well: https://forums.xamarin.com/discussion/comment/425482#Comment_425482

It seems i cant get the AVRouteDetector to initialize to set the [property](https://docs.microsoft.com/en-us/dotnet/api/avfoundation.avroutedetector.routedetectionenabled?view=xamarin-ios-sdk-12) RouteDetectionProperty to true to start scanning for Air Play devices for the AVRoutePickerView to work.

Ive also tried to use the notification and i dont get anything to come through there either. I was following this MPVolumeView to AVRoutePickerView [migration article](https://jeroenscode.com/avroutepickerview-replaces-mpvolumeview/).

**Comment by mandel-macaque** [comment]
@tj-devel709 has confirmed that the constructor is available. He will provide a fix and once it lands on main we will backport the fix to the branch for the next available release. 

Thanks for the detailed bug report!

**Comment by tj-devel709** [comment]
@gmoney494 
Hello, unfortunately, enabling the default constructor for AVRouteDetector caused a crash. Could you share your project so we can try to reproduce what you are using and thus look for a further solution?

**Comment by gmoney494** [comment]
@tj-devel709 im not allowed to share the project code. Do you require my VS and Xamarin.iOS set up? its in the original ticket above.

**Comment by tj-devel709** [comment]
It fails our tests when we allow the default constructor and we do not see anything that makes it seem like it should be user create-able. Would you be able to create a sample project just displaying what you are trying to do regarding the AVRouteDetector?

**Comment by gmoney494** [comment]
@tj-devel709 Ill work on trying to recreate how i use it since its in a custom renderer in Xamarin Forms. For now i have this code snippet from someone using in swift natively. 

`routeDetector = AVRouteDetector()
        guard let routeDetector = self.routeDetector else {
            return
        }
        routeDetector.isRouteDetectionEnabled = true`

**Comment by gmoney494** [comment]
Per the class definition, i dont see anything as to why it the default constructor wouldnt work. It has a base class of `NSObject`.

```
/**
 @class		AVRouteDetector
 @abstract	AVRouteDetector detects the presence of media playback routes.
 @discussion	
	If route detection is enabled (it is disabled by default), AVRouteDetector reports whether or not multiple playback routes have been detected. If this is the case, AVKit's AVRoutePickerView can be used to allow users to pick from the set of available routes.
 */
@available(iOS 11.0, *)
open class AVRouteDetector : NSObject {

    
    /**
     @property	routeDetectionEnabled
     @abstract	Whether or not route detection is enabled. The default value is NO.
     @discussion	
    	Route detection significantly increases power consumption and must be turned off when it's no longer needed.
     */
    open var isRouteDetectionEnabled: Bool

    
    /**
     @property	multipleRoutesDetected
     @abstract	This property is YES if, in addition to the local playback route, at least one more playback route has been detected.
     @discussion	
    	If multiple route have been detected, AVKit's AVRoutePickerView can be used to allow users to pick from the set of available routes. When the values of this property changes AVRouteDetectorMultipleRoutesDetectedDidChangeNotification is posted.
     */
    open var multipleRoutesDetected: Bool { get }
}
extension NSNotification.Name {

    /**
     @constant	AVRouteDetectorMultipleRoutesDetectedDidChangeNotification
     @abstract	Posted when the value of multipleRoutesDetected changes.
     */
    @available(iOS 11.0, *)
    public static let AVRouteDetectorMultipleRoutesDetectedDidChange: NSNotification.Name
}
```

**Comment by dalexsoto** [comment]
Hello @gmoney494 here is a workaround for your issue, this will allow you to call the parameterless ctor of  `AVRouteDetector`

```csharp
public class AVRouteDetectorExtensions
{
	static readonly IntPtr AllocSelector = ObjCRuntime.Selector.GetHandle ("alloc");
	static readonly IntPtr InitSelector = ObjCRuntime.Selector.GetHandle ("init");

	[DllImport (ObjCRuntime.Constants.ObjectiveCLibrary, EntryPoint = "objc_msgSend")]
	public extern static IntPtr IntPtr_objc_msgSend (IntPtr receiver, IntPtr selector);

	public static AVRouteDetector Create ()
	{
		IntPtr handle = IntPtr_objc_msgSend (ObjCRuntime.Class.GetHandle (typeof (AVRouteDetector)), AllocSelector);
		handle = IntPtr_objc_msgSend (handle, InitSelector);
		return ObjCRuntime.Runtime.GetNSObject<AVRouteDetector> (handle);
	}
}
```

### Usage

```csharp
var rd = AVRouteDetectorExtensions.Create ();
```

Please let us know if you are able to use this object as is in your application to evaluate if we should re-open https://github.com/xamarin/xamarin-macios/pull/10145. 

Thank you!

**Comment by gmoney494** [comment]
@dalexsoto So your workaround code works for me. I'm getting the `AVRouteDetector` to initialize properly and the events when the routes change. 

Only big issue I'm seeing now is accessing the `AVPlayer.externalPlaybackActive` property instead of `MPVolumeView.isWirelessRouteActive`. Since the native player and casting button are in seperate Renderer files to be used in Xamarin.Forms.

**Comment by dalexsoto** [comment]
> @dalexsoto So your workaround code works for me. I'm getting the AVRouteDetector to initialize properly and the events when the routes change.

Great, does this work on simulator and device? or just device ?

**Comment by rolfbjarne** [comment]
@dalexsoto @tj-devel709 I think we can just add the constructor, since it works for the customer in at least one of his scenarios.
